extensions/who-is-visiting-this-content/app.R

```diff
@@ -6,143 +6,355 @@ library(dplyr)
+              value = 0,
+              step = 0.5
+            ),
+
+            textInput(
+              "visit_lag_cutoff_text",
+              label = NULL,
+              value = 0
+            ),
+
+            actionButton("clear_cache", "Clear Cache", icon = icon("refresh"))
+          ),
+
+          div(
+            style = "display: flex; justify-content: space-between; gap: 1rem; align-items: center;",
+            div(
+              style = "display: inline-flex; gap: 0.5rem; align-items: center;",
+              h3(textOutput("content_title")),
+              uiOutput("dashboard_link")
+            ),
+            div(
+              style = "display: flex; gap: 0.5rem; align-items: center;",
+              textOutput("owner_info"),
+              uiOutput("email_owner_link")
+            )
+          ),
+
+          div(
+            style = "display: flex; justify-content: space-between; align-items: center;",
+            uiOutput("filter_message")
+          ),
+
+          layout_column_wrap(
+            width = "400px",
+            heights_equal = "row",
+            navset_card_tab(
+              tabPanel(
+                "Daily Visits",
+                div(
+                  style = "height: 300px",
+                  plotlyOutput("daily_visits_plot", height = "100%", width = "100%")
+                )
+              ),
+              tabPanel(
+                "Visit Timeline",
+                div(
+                  style = "height: 400px;",
+                  uiOutput("visit_timeline_ui")
+                )
+              )
+            ),
+
+            navset_card_tab(
+              tabPanel(
+                title = tagList(
+                  "Top Visitors",
+                  tooltip(
+                    bs_icon("info-circle-fill", class = "ms-2"),
+                    "Click a row to show only that user's visits."
+                  )
+                ),
+                reactableOutput("aggregated_visits")
+              ),
+              tabPanel(
+                "List of Visits",
+                reactableOutput("all_visits")
+              )
+            )
+          )
+        )
+      )
+    ),
+    tags$script(HTML("
+      document.addEventListener('DOMContentLoaded', function() {
+        const input = document.getElementById('guid_field');
+        if (input) {
+          input.addEventListener('keydown', function(e) {
+            if (e.key === 'Enter') {
+              e.preventDefault();
+              document.getElementById('submit_guid').click();
+            }
+          });
+        }
+      });
+    ")),
+    tags$style(HTML("
+      #selected_users + .selectize-control .selectize-input {
+        max-height: 150px;
+        overflow-y: auto;
+      }
+    "))
+  )
+}
 
-        textInput(
-          "visit_lag_cutoff_text",
-          label = "Visit Merge Window (sec)",
-          value = 1
-        ),
+server <- function(input, output, session) {
+  # Bookmarking ----
```

Comment:
Am I understanding the bookmarking correctly that it only works in "Open Solo" mode?

Reply:
I think so. There are also issues with some aspects of the bookmarking that I need to raise with the Shiny team, but haven't had time to do so yet.

Specifically, when updating a URL's query string via the default bookmarking pathway on Connect, it works fine, but when calling the `updateQueryString()` function manually, Connect adds an extra path segment to the URL. (https://positpbc.slack.com/archives/C0615R4HN9L/p1744902995229319)